components/fonts/platform/freetype/font.rs

```diff
@@ -128,23 +131,33 @@ impl PlatformFontMethods for PlatformFont {
             return Err("Could not memory map");
         };
 
-        // Variable fonts, where the font designer has provided one or more axes of
-        // variation do not count as font synthesis and their use is not affected by
+        // If the font face is already bold, applying synthetic bold would "double embolden" the font
+        let table_provider_data = FreeTypeFaceTableProviderData::Local(
+            Arc::new(memory_mapped_font_data),
+            font_identifier.index(),
+        );
+        let face_is_bold = table_provider_data
+            .font_ref()
+            .and_then(|font_ref| font_ref.os2())
+            .is_ok_and(|table| table.us_weight_class() >= SEMI_BOLD_U16);
+
+        // Variable fonts do not count as font synthesis and their use is not affected by
         // the font-synthesis property.
         //
         // <https://www.w3.org/TR/css-fonts-4/#font-synthesis-intro>
-        if FT_HAS_MULTIPLE_MASTERS(face.as_ptr()) {
-            synthetic_bold = false;
-        }
+        let is_variable_font = FT_HAS_MULTIPLE_MASTERS(face.as_ptr());
+
+        let synthetic_bold = if face_is_bold || is_variable_font {
+            false
+        } else {
+            synthetic_bold
+        };
```

Comment:
I think this is easier to understand as a single statement.
```suggestion
        let synthetic_bold = !face_is_bold && !is_variable_font && synthetic_bold;
```